lib/clientBuilder.js

```diff
@@ -113,6 +113,11 @@ function OktaAuthBuilder(args) {
     return window.location.hash;
   };
 
+  // This is exposed so we can mock setting window.location.hash in our tests
+  sdk.token.parseFromUrl._setLocationHash = function(hash) {
+    window.location.hash = hash;
```

Comment:
Am wondering if there's any value to using the History API instead, i.e. [this](http://stackoverflow.com/questions/1397329/how-to-remove-the-hash-from-window-location-with-javascript-without-page-refresh/5298684#5298684) with replaceState instead of pushState.